forecasting/src/autogluon/forecasting/dataset/ts_dataframe.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import annotations
+
+from typing import Any, Tuple, Dict, List
+from dataclasses import dataclass
+
+import pandas as pd
+import numpy as np
+
+
+@dataclass
+class TimeSeriesListDataset:
+    data_iter: List[Dict[str, Any]]
+    freq: str
+
+
+class TimeSeriesDataFrame(pd.DataFrame):
+    """TimeSeriesDataFrame to represent time-series dataset.
+
+    Parameters
+    ----------
+    data : Any
+        Time-series data to construct a TimeSeriesDataFrame.
+        It currently supports two formats:
+            1. Time-series data in TimeSeriesListDataset format.
+            2. Time-series data pd.DataFrame format with multi-index on item and timestamp.
+            An example can be found using example_dataframe() function.
+    """
+
+    def __init__(self, data: Any, *args, **kwargs):
+        if isinstance(data, TimeSeriesListDataset):
+            data = self.from_list_dataset(data)
```

Comment:
Couldn't we implement `as_dataframe` on `TimeSeriesListDataset` instead?

Reply:
I think we never want to instantiate / pass around `TimeSeriesListDatasets`, but this just serves as a Type. So I agree with Jasper's point on `UserList`, but not with implementing `as_dataframe`. `TimeSeriesListDataset` will never be depended on by the main library.

Reply:
Yes, as Caner told, the core data structure used in the forecasting module will be `TimeSeriesDataFrame`, the users does not need to know `TimeSeriesListDataset`.